app/integration/api/src/main/java/io/syndesis/integration/api/IntegrationResourceManager.java

```diff
@@ -202,10 +208,21 @@ default Integration sanitize(Integration integration) {
         }
 
         return new Integration.Builder().createFrom(integration)
+            .name(sanitizeIntegrationName)
             .flows(replacementFlows)
             .build();
     }
 
+    /**
+     * Sanitize an integration name according a specific set of characters
+     *
+     * @param name the name to sanitize
+     * @return a sanitized name replacing illegal characters
+     */
+    default String sanitize(String name) {
+        return name == null ? null : Names.sanitize(name);
```

Comment:
I'd either move this to `Names` or make it throw NullPointerException, I think integration name is required (not sure how enforced though)

Reply:
I partially agree on that. Actually the `sanitize()` method is meant to be generic, not necessarily specific to the name. We already have in place a validation of the `null` value in the model https://github.com/syndesisio/syndesis/blob/39513aa3af7ab70633f31875ec2d155651abeb07/app/common/model/src/main/java/io/syndesis/common/model/WithName.java#L28
Since it's something generic I prefer to have the check since I don't know before hand what kind of values will be provided there.